### PR TITLE
OEL-1463: Update pager test assertions for first/last.

### PR DESCRIPTION
**Pull request by donquixote**

### CODE REVIEW

File: modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php [hunk 1]
```diff
@@ -225,32 +225,54 @@ protected function assertSearchResults(array $expected): void {
...
+
+    $get_page_index = static function (NodeElement $link): string {
+      $href = $link->getAttribute('href');
+      $query_string = parse_url($href, PHP_URL_QUERY);
+      parse_str($query_string, $query);
+      return $query['page'];
+    };
+
+    if ($active === $total - 1) {
+      $active_link = $pagination->find('css', 'li.active:last-child > a');
```
Comment (drishu) [inline modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php]: isn't it sufficient to call li.active to fetch the active link?

File: modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php [hunk 2]
```diff
@@ -225,32 +225,54 @@ protected function assertSearchResults(array $expected): void {
...
+      $last_link = NULL;
```
Comment (drishu) [inline modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php]: ->find returns null if not found no ?
combined with above comment I think we can skip the if else and just grap the 2 things, then check for null or something

File: modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php [hunk 3]
```diff
@@ -225,32 +225,54 @@ protected function assertSearchResults(array $expected): void {
...
    */
-  protected function assertPagerLinks(array $expected, int $active_item_index = 0): void {
-    if ($expected === []) {
-      $this->assertSession()
-        ->elementNotExists('css', 'ul.pagination');
+  protected function assertPager(int $total, int $active = 0): void {
+    $pagination = $this->assertSession()
+      ->elementExists('css', 'ul.pagination');
+
+    $get_page_index = static function (NodeElement $link): string {
```
Comment (drishu) [inline modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php]: I agree that I would have also put it in a helper function, named or not, both are fine for me here, but do try to stick to certain scenarios when using them:
1. when there is no reusability within your project/you use it only in that specific place
2. as value of a callable parameter

at least this is my compass for it

**Review by drishu — CHANGES_REQUESTED** [review]

File: modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php (hunk 1 above)
Comment (donquixote) [reply modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php]: in this case the active and last link are identical. i want to verify that this is the case.

Comment (drishu) [reply modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php]: you check the page index of the active link and then separately  for the last link, I dont see why you would care if its the same, but I wont insist, end result is the same so up 2 u

File: modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php [hunk 4]
```diff
@@ -225,32 +225,65 @@ protected function assertSearchResults(array $expected): void {
...
+
+  /**
+   * Asserts the page index in a pager link.
+   *
+   * @param int $expected
+   *   Expected page number.
+   * @param \Behat\Mink\Element\NodeElement $pager_link
+   *   A link element that belongs to a pager.
+   */
+  protected function assertPagerLinkPageNumber(int $expected, NodeElement $pager_link): void {
```
Comment (drishu) [inline modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php:265]: I would have expected a string and typecasted when sending the value but not a biggy

**Review by drishu — DISMISSED** [review]

File: modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php (hunk 4 above)
Comment (donquixote) [reply modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php:265]: I'd say semantically it is a number, and the `int` is more restrictive so gives a better idea what kind of value we expect.

File: modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php (hunk 1 above)
Comment (donquixote) [reply modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php]: I agree now.
My earlier motivation was that I don't fully trust the pager, and wanted to be sure the last item is really the last link with the icon, not a bug where the last link is missing for some reason. This is why I wanted to check labels _and_ link destination.
But we cannot reliably check this anyway. So now I drop the label checks.

File: modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php (hunk 4 above)
Comment (donquixote) [reply modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php:265]: Also I wanted to get the typecast out of the calling code, e.g. no `(string) ($total - 1)`.

**Review by drishu — DISMISSED** [review]

**Review by escuriola — DISMISSED** [review]

**Review by brummbar — DISMISSED** [review]

**Comment by donquixote** [comment]
Pushed again, removed the commit in composer.json

**Review by drishu — APPROVED** [review]

**Review by brummbar — APPROVED** [review]

**Pull request merged**